app/controllers/challenge/gift_exchange_controller.rb

```diff
@@ -50,7 +50,7 @@ def update
       flash[:notice] = ts('Challenge was successfully updated.')
 
       # expire the cache on the signup form
-      expire_fragment(:controller => 'challenge_signups', :action => 'new')
+      ActionController::Base.new.expire_fragment('challenge_signups/new')
```

Comment:
Upgrading to Rails 4 causes an exception: `undefined method 'expire_fragment'` that is solved by adding the `ActionController::Base.new` prefix. A number of relevant threads, but this one is concise: http://stackoverflow.com/questions/33499384/expire-fragment-caching-in-rails-4